airbyte_cdk/entrypoint.py

```diff
@@ -141,19 +141,33 @@ def run(self, parsed_args: argparse.Namespace) -> Iterable[str]:
                 )
                 if cmd == "spec":
                     message = AirbyteMessage(type=Type.SPEC, spec=source_spec)
-                    yield from [
+                    yield from (
                         self.airbyte_message_to_string(queued_message)
                         for queued_message in self._emit_queued_messages(self.source)
-                    ]
+                    )
                     yield self.airbyte_message_to_string(message)
+                elif (
+                    cmd == "discover"
+                    and not parsed_args.config
+                    and not self.source.check_config_against_spec
+                ):
+                    empty_config: dict[str, Any] = {}
```

Comment:
FYI - I'm inserting a comment to help the reader.

```suggestion
                    # Connector supports unprivileged discover
                    empty_config: dict[str, Any] = {}
```